DA-190: Increase performance of projects, documents and scheme requests

JeannedArk opened the pull request:
The entity mapping and queries were improved so that the db requests
are significantly faster and do not create unnecessary queries.
Almost all FetchTypes were set to LAZY and native queries were
rewritten in JPQL named queries. Query hints were added to prevent
unnecessary queries.
rootController was set for the dashboard because the profileController
requested unnecessarily timelogging.

JeannedArk commented:
@stgrue or @judembo or @cbocionek pls test

judembo commented:
There seems to be a problem with creating projects. While projects that were previously in the database (created on master branch) are displayed correctly, newly created projects are not visible (neither to admin nor to annotator) after logging out or changing to another page, although they are stored in the database. Also, adding documents to a project doesn't work for me.

JeannedArk committed:
DA-190: Increase performance of projects, documents and scheme requests

The entity mapping and queries were improved so that the db requests
are significantly faster and do not create unnecessary queries.
Almost all FetchTypes were set to LAZY and native queries were
rewritten in JPQL named queries. Query hints were added to prevent
unnecessary queries.
rootController was set for the dashboard because the profileController
requested unnecessarily timelogging.
